sklearn/utils/stats.py

```diff
@@ -1,18 +1,46 @@
 import numpy as np
 
 from .extmath import stable_cumsum
+from sklearn.utils.fixes import _take_along_axis
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
     """
     Compute the weighted ``percentile`` of ``array`` with ``sample_weight``.
+    If ``array`` is 2D, compute weighted ``percentile`` along axis=0.
     """
-    sorted_idx = np.argsort(array)
+    if sample_weight is None:
+        sample_weight = np.ones_like(array)
```

Comment:
Depending on whether this slows down the current version, we might want to have 2 versions: one when sample_weight is None, and one when it's not None.

Maybe wait until your code is mature to benchmark this though.

Reply:
Oh I did what you suggested and used `np.median()` when `sample_weight=None`. I guess I can remove this. I think I kept it just in case/for robustness but since this is an internal function will remove.